3rd-party/libssh/CMakeLists.txt

```diff
@@ -41,8 +41,43 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR}/libssh)
 include(libssh/ConfigureChecks.cmake)
 configure_file(libssh/config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/libssh/config.h)
 
-# We bypass the main CMake file to avoid various unneded checks
-add_subdirectory(libssh/src EXCLUDE_FROM_ALL)
+# Must be set after configure checks as crypto has not been built yet
+# but will be included in the final libssh shared library
+set(OPENSSL_CRYPTO_LIBRARY crypto decrepit ssh-boringssl-compat)
+
+# Since the main CMake file is bypassed, we have to search that CMake file to determine the proper
+# shared library versions
+file(STRINGS libssh/CMakeLists.txt libssh_version_line REGEX ".*LIBRARY_VERSION.*")
+file(STRINGS libssh/CMakeLists.txt libssh_soversion_line REGEX ".*LIBRARY_SOVERSION.*")
+
+if (NOT libssh_version_line)
+    message(FATAL_ERROR "no library version specified in libssh project")
+endif()
+
+if (NOT libssh_soversion_line)
+    message(FATAL_ERROR "no library soversion specified in libssh project")
+endif()
+
+string(REPLACE "\"" "" libssh_version_line ${libssh_version_line})
+string(REPLACE "\"" "" libssh_soversion_line ${libssh_soversion_line})
+
+unset(CMAKE_MATCH_1)
+string(REGEX MATCH ".*LIBRARY_VERSION (.*)\\)" _ ${libssh_version_line})
+if (NOT CMAKE_MATCH_1)
+    message(FATAL_ERROR "unable to find libssh library version")
+endif()
+set(LIBRARY_VERSION ${CMAKE_MATCH_1})
+
+unset(CMAKE_MATCH_1)
+string(REGEX MATCH ".*LIBRARY_SOVERSION (.*)\\)" _ ${libssh_soversion_line})
+if (NOT CMAKE_MATCH_1)
+    message(FATAL_ERROR "unable to find libssh library soversion")
+endif()
+set(LIBRARY_SOVERSION ${CMAKE_MATCH_1})
```

Comment:
I PR'd a simplification of the above, wdyt? albaguirre#1

Reply:
LGTM